.github/workflows/build.yml

```diff
@@ -34,13 +34,13 @@ on:
 jobs:
   build_cli:
     name: Build CLI
-    runs-on: ${{matrix.os}}
+    runs-on: ${{matrix.os.name}}
     strategy:
       matrix:
-        os: [ubuntu-20.04, windows-2022, macos-12]
+        os: [{ name: ubuntu-20.04, bin-name: linux }, { name: windows-2022, bin-name: windows }, { name: macos-12, bin-name: darwin }]
         arch: [amd64, arm64]
         exclude:
```

Comment:
Add an exclusion for macos-12-amd64 as well? It's important to run tests for this combination, but we can save on build time.